main/src/library/List.flix

```diff
@@ -0,0 +1,527 @@
+/*
+ * Copyright 2016 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace List {
+
+    ///
+    /// Returns true iff `xs` is the empty list, i.e. `Nil`.
+    ///
+    def null[a](xs: List[a]): Bool = match xs with {
```

Comment:
Another note about naming convention: Haskell typically uses all lowercase names, while Scala will use camelCase. What do we want in Flix?